src/App.tsx

```diff
@@ -1,57 +1,123 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function debounce<T extends any[]>(func: (...args: T) => void, delay: number) {
+  let timeoutId: NodeJS.Timeout;
 
-  return (
-    <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+  // eslint-disable-next-line func-names
+  return function (...args: T) {
+    clearTimeout(timeoutId);
+    timeoutId = setTimeout(() => {
+      func(...args);
+    }, delay);
+  };
+}
+
+const Autocomplete: React.FC<{
+  debounceDelay: number;
+  onSelected: (person: Person | null) => void }>
+  = ({ debounceDelay, onSelected }) => {
+    const [searchText, setSearchText] = useState<string>('');
```

Comment:
In this case, it is not necessary to explicitly indicate the type of the variable, because it is determined by the starting value